Draft content for Combat chapter (#9)
joeabolt/DawnfireRPG: Combat.md

# Combat

> Combat is a unfortunate reality among those who specialize in dealing with dangerous situations. Whether you have to fend off the cultists who are suspicious you know too much or accidentally got caught up slaying dragons, you will eventually find yourself trying to kill or incapacitate your opponents before they kill or incapacitate you. That is what this chapter is for.

Author: ```suggestion
Combat is an unfortunate reality among those who specialize in dealing with dangerous situations. Whether you must fend off cultists who are suspicious you know too much or accidentally get caught up slaying dragons, you will eventually find yourself trying to kill or incapacitate your opponents before they kill or incapacitate you. Hence, this chapter.
```

## Overview

> Combat is organized into [Rounds]. One [Round] is the time it takes everyone to take one turn in combat (in-game, it is between 2 and 10 seconds, usually 6). On your turn each [Round], you will lose any Action Points (AP) you previously had and receive 3 new AP that you can spend taking actions. When everyone's turn is over, a new [Round] begins.

Author: ```suggestion
Combat is organized into [Rounds]. One [Round] is the time it takes everyone to take one turn in combat (in-game, about 6 seconds). On your turn each [Round], you will lose any Action Points (AP) you previously had and receive 3 new AP that you can then spend taking actions. When everyone's turn is over, a new [Round] begins.
```

> ### Initiative and Turn Order

Author: Add optional rule for team-based initiative. Probably something like "have the lowest init bonus roll" or something like that.

When deciding who goes first in a [Round], have everyone roll 1d20 and add their initiative. By default, everyone takes actions in order from highest to lowest - break ties with a re-roll.

> You can always choose to treat your initiative as lower than what you rolled.

Author: Move this line into the previous paragraph, and specify that order is maintained throughout combat.

> You and an ally can swap positions in the initiative order at the beginning of combat. Whoever rolled higher takes the initiative score of the other. Whoever rolled lower takes the higher initiative score - 2, or their initiative score + 1, whichever is lower. _Example: Alice rolls a 17 and Bob rolls a 5. They swap, so Alice has a 5 and Bob has a 15. Example: Alice rolls a 12 and Bob rolls a 10. They swap. Alice has a 10 and Bob has an 11._

Author: Clarify rules around delaying.

### On Your Turn
At the beginning of your turn, you lose any AP you did not use last [Round] and gain 3 new AP. You can then begin taking actions, spending AP as appropriate. The following table lists action types and their AP cost. The next section lists some common combat actions.

> 

Author: Insert note about doing non-common combat actions. I.e., how intensive does an action need to be to be Instant/Fast/Standard/Slow. (Talking is an [Instant Action]! (but with realism, it's a [Fast Action]))

|Action Type|AP Cost|
|---|---|
|[Instant Action]|0 AP|
|[Fast Action]|1 AP|
|[Standard Action]|2 AP|
|[Slow Action]|3 AP|
|[Immediate Action]|0 AP*|
|[Reaction]|0 AP*|

> __[Immediate Actions] and [Reactions]:__ These special kinds of actions can be taken when it is not your turn. Anything that is an [Immediate Action] or [Reaction] will list the precise conditions under which it can be taken. While you can take any number of [Immediate Actions], many GMs will limit this to a reasonable number (say, 3). However, you can only ever take a single [Reaction] per [Round]. 

Author: ```suggestion
__[Immediate Actions] and [Reactions]:__ These special kinds of actions can be taken when it is not your turn. Anything that is an [Immediate Action] or [Reaction] will list the precise conditions under which it can be taken. You can only ever take a single [Reaction] per [Round]. Though you can take any number of [Immediate Actions] per [Round], many GMs will limit this to a reasonable number (say, 3).
```

## Common Actions

### Standard Attack

> Also known as "attack." This is a [Standard Action]. Select one of your weapons you are currently wielding and an opponent you can target with that weapon. Make an attack roll. If you are using a martial weapon, you have to equal or exceed their AC to hit. If you are using a magical weapon, you have to equal or exceed their SR to hit (and use a spell attack roll).

Author: ```suggestion
Also known as "attack." This is a [Standard Action]. Select one of your weapons you are currently wielding and an opponent you can target with that weapon. Make an attack roll. With a martial weapon, you must equal or exceed their AC to hit. With a magical weapon, you must equal or exceed their SR, and use a spell attack roll.
```

_Attack roll = 1d20 + Strength + weapon bonuses + misc. bonuses_  
_Spell Attack Roll = 1d20 + Intelligence + weapon bonuses + misc. bonuses_

> If the natural result of the d20 - the number that lands facing up - is equal to or greater than your weapons critical threshold, you scored a critical hit. If the attack roll still was not high enough to hit your opponent, you still hit but it is only a regular hit.

Author: ```suggestion
If the natural result of the d20 - the number that lands facing up - is equal to or greater than your weapon's critical threshold, you scored a critical hit. If the attack roll was not high enough to hit your opponent, you miss. If your attack roll was not high enough but was in your critical range, you land a regular hit - not a critical hit.
```

> If your opponent currently has any [Miss Chances], check those now. Roll d%. If you roll less than or equal to their [Miss Chance], you hit. Otherwise, you miss. If you have a critical hit but would miss, you still hit but it is only a regular hit.

Author: ```suggestion
If your opponent currently has any [Miss Chances], check those now. Roll d%. If you roll less than or equal to their [Miss Chance], you hit. Otherwise, you miss. If you have a critical hit that would miss from [Miss Chances], it becomes a regular hit instead of a miss.
```

> Then, you roll damage, based on the weapon. If a critical hit, multiply this by your weapon's critical multiplier. Then apply any [Weakness] or [Resistance] the target has. Then they subtract the damage from their health.

Author: ```suggestion
Then, roll damage as per your weapon. If a critical hit, multiply this total by your weapon's critical multiplier. Then apply any [Weakness] or [Resistance] the target has. Finally, the target subtracts the damage from their health.
```

All told, the process looks like this:
1. Choose weapon and target
1. Attack roll
1. Miss Chance
1. Roll base damage
1. Apply critical multiplier, [Weakness], and [Resistance]
1. Target takes damage

### Standard Defense
A favorite of pacifists and defensive fighters. This involves using your weapon to interrupt enemy attacks, getting your shield in the way, or otherwise focusing on defense. This is a [Standard Action]. If you have a melee weapon, you may make an attack roll. Otherwise, roll 1d20 + initiative + one-half your level. Your AC and SR receive a +4 bonus until your next turn, but cannot exceed the results of this roll. This can only be used once per [Round].

### Move
This is a [Standard Action], allowing you to move up to your movement speed. When moving on a square grid, the first diagonal step as part of this movement counts as 5 feet, and every diagonal step after counts as 10 feet.

### 5-Foot Step
This is a [Fast Action]. You move 5 feet in any direction. You may only take one 5-foot step per turn.

> ### Cast a Spell

Author: Clarify multi-casting rules, as well as the requirement to advance all castings every turn (but option to advance multiple times in one turn).

This is a [Fast Action] that advances a spell's casting by 1 AP. When you reduce the time needed to cast the spell to 0 AP, you may cast the spell as an [Instant Action] (some spells instead let you release the spell later, and you can always choose to let a spell fizzle). To finish casting the spell, you must immediately pay the spell's casting cost.

> ## Combat Maneuvers

Author: Need to add Run as a combat maneuver ([Slow Action], move double speed).

### Quick Attack
This is a [Fast Action]. Make an attack. If the attack roll hits, the attack automatically does minimum damage.

### Quick Defense
This is a [Fast Action]. It works the same as Standard Defense, except your AC and SR can only go up by +2. This stacks with Standard Defense, but not with itself.

### Power Attack
This is a [Slow Action]. Make an attack. If the attack roll hits, you deal normal damage but roll the weapon's damage dice twice.

Example: The damage dice for a Longsword are 2d8, with Strength and one-half the wielder's level added to damage. On a Power Attack, the wielder rolls 4d8, adding their Strength and one-half their level.

## Optional Rules
These rules add more realism to combat, but they also add more bookkeeping, and are not needed to have a good time. As such, they are __strictly optional.__

### Reach

> These rules make it easier to strike an opponent when your weapon is longer than theirs. It applies only to melee weapons.

Author: ```suggestion
These rules make it easier to strike an opponent in melee when your weapon is longer than theirs. It applies only to melee weapons.
```
